copybookreader-plugins/src/test/java/co/cask/hydrator/plugin/CopyBookTest.java

```diff
@@ -0,0 +1,224 @@
+  private static final ArtifactSummary ETLBATCH_ARTIFACT =
+    new ArtifactSummary(BATCH_APP_ARTIFACT_ID.getArtifact(), BATCH_APP_ARTIFACT_ID.getVersion());
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private String cblContents = "000100*                                                                         \n" +
+    "000200*   DTAR020 IS THE OUTPUT FROM DTAB020 FROM THE IML                       \n" +
+    "000300*   CENTRAL REPORTING SYSTEM                                              \n" +
+    "000400*                                                                         \n" +
+    "000500*   CREATED BY BRUCE ARTHUR  19/12/90                                     \n" +
+    "000600*                                                                         \n" +
+    "000700*   RECORD LENGTH IS 27.                                                  \n" +
+    "000800*                                                                         \n" +
+    "000900        03  DTAR020-KCODE-STORE-KEY.                                      \n" +
+    "001000            05 DTAR020-KEYCODE-NO      PIC X(08).                         \n" +
+    "001100            05 DTAR020-STORE-NO        PIC S9(03)   COMP-3.               \n" +
+    "001200        03  DTAR020-DATE               PIC S9(07)   COMP-3.               \n" +
+    "001300        03  DTAR020-DEPT-NO            PIC S9(03)   COMP-3.               \n" +
+    "001400        03  DTAR020-QTY-SOLD           PIC S9(9)    COMP-3.               \n" +
+    "001500        03  DTAR020-SALE-PRICE         PIC S9(9)V99 COMP-3.";
+
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+
+    // add artifact for batch sources and sinks
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("copybookreader-plugins", "1.0.0"), BATCH_APP_ARTIFACT_ID,
+                      CopybookSource.class);
+
+    FileInputFormat.setInputPaths(new JobConf(), new Path("src/test/resources"));
+  }
+
+  @Test
+  public void testCopybookReaderWithOutputSchema() throws Exception {
+
+    Schema schema = Schema.recordOf(
+      "event",
+      Schema.Field.of("DTAR020-KEYCODE-NO", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("DATE", Schema.nullableOf(Schema.of(Schema.Type.INT))),
+      Schema.Field.of("DTAR020-DEPT-NO", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("DTAR020-QTY-SOLD", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("DTAR020-SALE-PRICE", Schema.of(Schema.Type.DOUBLE)));
+
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put(Constants.Reference.REFERENCE_NAME, "TestCass")
+      .put("binaryFilePath", "src/test/resources/DTAR020_FB.bin")
+      .put("copybookContents", cblContents)
+      .put("schema", schema.toString())
+      .put("fileStructure", "")
```

Comment:
since this is nullable, we can omit this attribute here, right? (as opposed to specifying empty string).